src/compiler/parser.ts

```diff
@@ -17,6 +17,13 @@ namespace ts {
     let IdentifierConstructor: new (kind: SyntaxKind, pos: number, end: number) => Node;
     let SourceFileConstructor: new (kind: SyntaxKind, pos: number, end: number) => Node;
 
+    interface Fail { __FAIL: void; }
+    /** Only value of type Fail. */
+    const Fail: Fail = { __FAIL: undefined };
+    function isFail(x: {} | null | undefined | void): x is Fail {

```

Comment:
How is `{} | null | undefined | void` really any different from `any` in this case? This seems unnecessarily verbose.